docs/concepts/ingress-classes.md

```diff
@@ -0,0 +1,181 @@
+# Kong Ingress Controller and ingress class
+
+
+## Table of Contents
+
+- [**Introduction**](#introduction)
+- [**Configuring the controller class**](#configuring-the-controller-ingress-class)
+- [**Loading resources by class**](#loading-resouces-by-class)
+- [**When to use a custom class**](#when-to-use-a-custom-class)
+- [**Legacy behavior**](#legacy-behavior)
+- [**Examples**](#examples)
+
+## Introduction
+
+The Kong Ingress Controller uses ingress classes to filter Kubernetes Ingress
+objects and other resources before converting them into Kong configuration.
+This allows it to coexist with other ingress controllers and/or other
+deployments of the Kong Ingress Controller in the same cluster: a Kong Ingress
+Controller will only process configuration marked for its use.
+
+## Configuring the controller ingress class
+
+The `--ingress-class` flag (or `CONTROLLER_INGRESS_CLASS` environment variable)
+specify the ingress class expected by the Kong Ingress Controller. By default,
+it expects the `kong` class.
+
+## Loading resources by class
+
+The Kong Ingress Controller translates a variety of Kubernetes resources into
+Kong configuration. Broadly speaking, we can separate these resources into two
+categories:
+
+- Resources that the controller translates directly into Kong configuration.
+- Resources referenced by some other resource, where the other resource is
+  directly translated into Kong configuration.
+
+For example, an Ingress is translated directly into a Kong route, and a
```

Comment:
Unnecessary here. The focus of this content isn't to discuss basic Kong proxy concepts, and we may reasonably assume many readers are familiar with them: they're part of the basic set of configuration introduced in our getting started guides.

Reply:
Three arguments against not including it here:
1. It costs nothing to add (because the link just changes a term into a hyperlink, and the likelihood of the link rotting is low)
2. 
    > we may reasonably assume many readers are familiar with them

    This is a harmful and (imo unnecessary) assumption: some people will come here from a Google search, some people will be only 90% sure about the relationships between routes, services and upstreams.
3. Wikipedia would cross-link in such situation. Take for example the [Molotov-Ribbentrop pact](https://en.wikipedia.org/wiki/Molotov%E2%80%93Ribbentrop_Pact) article. The reader may be reasonably assumed to be familiar with terms "Soviet Union" and "Germany", but the article has them linked nonetheless.